src/prommis/uky/costing/ree_plant_capcost.py

```diff
@@ -296,648 +454,673 @@ def build_process_costs(
+                    doc="Project management and construction",
+                )
 
-            self.piping_materials_and_labor_costs = Var(
-                initialize=value(self.total_BEC),
-                bounds=(0, 1e4),
-                doc="Piping, materials and labor ancillary cost in $MM",
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
+                self.process_contingency_percentage = Param(
+                    mutable=True,
+                    initialize=process_contingency_percentage / 100,
+                    doc="Process contingency",
+                )
 
-            self.electrical_materials_and_labor_costs = Var(
-                initialize=value(self.total_BEC),
-                bounds=(0, 1e4),
-                doc="Electrical, materials and labor ancillary cost in $MM",
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
+                # ancillary cost variables
+                self.ancillary_costs = Var(
+                    initialize=value(self.total_BEC),
+                    bounds=(0, 1e4),
+                    doc="Ancillary cost in $MM",
+                    units=getattr(pyunits, "MUSD_" + CE_index_year),
+                )
 
-            self.instrumentation_costs = Var(
-                initialize=value(self.total_BEC),
-                bounds=(0, 1e4),
-                doc="Ancillary cost in $MM",
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
+                self.piping_materials_and_labor_costs = Var(
+                    initialize=value(self.total_BEC),
+                    bounds=(0, 1e4),
+                    doc="Piping, materials and labor ancillary cost in $MM",
+                    units=getattr(pyunits, "MUSD_" + CE_index_year),
+                )
 
-            self.plant_services_costs = Var(
-                initialize=value(self.total_BEC),
-                bounds=(0, 1e4),
-                doc="Ancillary cost in $MM",
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
+                self.electrical_materials_and_labor_costs = Var(
+                    initialize=value(self.total_BEC),
+                    bounds=(0, 1e4),
+                    doc="Electrical, materials and labor ancillary cost in $MM",
+                    units=getattr(pyunits, "MUSD_" + CE_index_year),
+                )
 
-            # buildings cost variables
-            self.buildings_costs = Var(
-                initialize=value(self.total_BEC),
-                bounds=(0, 1e4),
-                doc="Buildings cost in $MM",
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
+                self.instrumentation_costs = Var(
+                    initialize=value(self.total_BEC),
+                    bounds=(0, 1e4),
+                    doc="Ancillary cost in $MM",
+                    units=getattr(pyunits, "MUSD_" + CE_index_year),
```

Comment:
Rather than build the units string each time you use it, just build it once up front as a local object and use it wherever it is needed. Otherwise, you are duplicating operations and wasting CPU time.

Also, what happens if the user wants to use Euros or some other currency units?

Reply:
See answer below.